Make requirements for data module optional (#3257)
PennyLaneAI/pennylane: doc/introduction/data.rst

PennyLane provides the :mod:`~.data` subpackage to download, create, store and manipulate quantum datasets.

.. note::

>     For using :mod:`~pennyalane.data` module, users would require `zstd` and `dill` packages.

Reviewer: ```suggestion
    The packages `zstd` and `dill` are required to use the :mod:`~pennyalane.data` module.
```
